tailmac/Swift/Common/Config.swift

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+import Foundation
+
+
+/// Represennts a configuration for a particular virutal machine
+class Config: Codable {
+    var serverSocket = "/tmp/qemu.sock"
+    var memorySize = (4 * 1024 * 1024 * 1024) as UInt64
+    var mac = "5a:94:ef:e4:0c:ee"
+    var port: UInt32 = 51009
+
+    // The virtual machines ID.  Also double as the directory name under which
+    // we will store configuration, block device, etc.
+    let vmID: String
+
+    init(_ vmID: String = "default") {
+        self.vmID = vmID
+        let configFile = vmDataURL.appendingPathComponent("config.json")
+        if FileManager.default.fileExists(atPath: configFile.path()) {
+            print("Using config file at path \(configFile)")
+            if let jsonData = try? Data(contentsOf: configFile) {
+                let config = try! JSONDecoder().decode(Config.self, from: jsonData)
+                self.serverSocket = config.serverSocket
+                self.memorySize = config.memorySize
+                self.mac = config.mac
+                self.port = config.port
+            }
+        }
+    }
+
+    func persist() {
+        let configFile = vmDataURL.appendingPathComponent("config.json")
+        let data = try! JSONEncoder().encode(self)
+        try! data.write(to: configFile)
+    }
+
+    lazy var restoreImageURL: URL = {
+        vmBundleURL.appendingPathComponent("RestoreImage.ipsw")
+    }()
+
+    // The VM Bundle URL holds the restore image and a set of VM images
+    // By default, VM's are persisted at ~/VM.bundle
+    lazy var vmBundleURL: URL = {
+        let vmBundlePath = NSHomeDirectory() + "/VM.bundle/"
+        createDir(vmBundlePath)
+        let bundleURL = URL(fileURLWithPath: vmBundlePath)
+        print("Starting VM with base path \(bundleURL)")
+        return bundleURL
+    }()
+
+    // The VM Data URL holds the specific files composing a unique VM guest instance
+    // By default, VM's are persisted at ~/VM.bundle/<vmID>
+    lazy var vmDataURL =  {
+        let dataURL = vmBundleURL.appendingPathComponent(vmID)
+        print("Using VM instance path \(dataURL)")
+        return dataURL
+    }()
+
+    lazy var auxiliaryStorageURL = {
+        vmDataURL.appendingPathComponent("AuxiliaryStorage")
+    }()
+
+    lazy var diskImageURL = {
+        vmDataURL.appendingPathComponent("Disk.img")
+    }()
+
+    lazy var hardwareModelURL = {
+        vmDataURL.appendingPathComponent("HardwareModel")
+    }()
+
+    lazy var machineIdentifierURL = {
+        vmDataURL.appendingPathComponent("MachineIdentifier")
+    }()
+
+    lazy var saveFileURL = {
+        vmDataURL.appendingPathComponent("SaveFile.vzvmsave")
+    }()
+
+    private func createDir(_ path: String) {
+        do {
+            try  FileManager.default.createDirectory(atPath: path, withIntermediateDirectories: true)
+        } catch {
+            fatalError("Unable to create VM images/assets dir at \(path)")
```

Comment:
Let's print the error:

```suggestion
            fatalError("Unable to create VM images/assets dir at \(path): \(error)")
```